api/src/backend/api/models.py

```diff
@@ -187,12 +187,14 @@ def validate_gcp_uid(value):
 
     @staticmethod
     def validate_kubernetes_uid(value):
-        if not re.match(r"^[a-z0-9]([-a-z0-9]{1,61}[a-z0-9])?$", value):
-            raise ModelValidationError(
-                detail="K8s provider ID must be up to 63 characters, start and end with a lowercase letter or number, "
-                "and contain only lowercase alphanumeric characters and hyphens.",
-                code="kubernetes-uid",
-                pointer="/data/attributes/uid",
+        if not re.match(
+            r"(^[a-z0-9]([-a-z0-9]{1,61}[a-z0-9])?$)|(^arn:aws(-cn|-us-gov|-iso|-iso-b)?:[a-zA-Z0-9\-]+:([a-z]{2}-[a-z]+-\d{1})?:(\d{12})?:[a-zA-Z0-9\-_\/:\.\*]+(:\d+)?$)",
+            value,
+        ):
+            raise ValueError(
+                "The value must either be a valid Kubernetes UID (up to 63 characters, "
+                "starting and ending with a lowercase letter or number, containing only "
+                "lowercase alphanumeric characters and hyphens) or a valid EKS ARN."
```

Comment:
Please, update regexes and detail messages, but don't modify the current logic. Use `ModelValidationError` with all the required fields. This breaks this exception.

Reply:
Good catch, I will change it now!